### PR TITLE
New cat option

### PR DESCRIPTION
**Pull request by yishain11**
task #97

### CODE REVIEW

File: frontend/src/contexts/group.tsx [hunk 1]
```diff
@@ -67,9 +69,10 @@ const GroupProvider: FC = () => {
     id: currentGroup?.id || -1,
     activeAgreements: activeAgreements?.core_agreements || [],
     archivedAgreements: archivedAgreements?.core_agreements || [],
-    categories: categoriesData?.core_categories || [],
+    categories: categoriesData?.core_categories || [{ name: 'New Category', id: NaN }],
```
Comment (avirankatz) [inline frontend/src/contexts/group.tsx:72]: New category shouldn't be included in group's categories

File: frontend/src/contexts/group.tsx [hunk 2]
```diff
@@ -67,9 +69,10 @@ const GroupProvider: FC = () => {
...
     archiveAgreement: (id, iArchived) => archiveAgreement({ variables: { id, iArchived } }),
   };
+  state.categories = [...state.categories, { name: 'New Category', id: NaN }];
```
Comment (avirankatz) [inline frontend/src/contexts/group.tsx:75]: New category shouldn't be included in group's categories

File: frontend/src/pages/NewAgreement.tsx [hunk 3]
```diff
@@ -62,6 +62,7 @@ const NewAgreement: FC = () => {
    * Category
    */
   const [categoryId, setCategoryId] = useState<number | null>(null); //@todo set default based on extant record in Hasura
+  console.log('categoryId', categoryId);
```
Comment (avirankatz) [inline frontend/src/pages/NewAgreement.tsx:65]: Let's remove this print

File: frontend/src/components/CategorySelect.tsx [hunk 4]
```diff
@@ -112,11 +121,24 @@ const CategorySelect: FC<ICategorySelectProps> = ({
           '& .MuiSelect-icon': { color: '#f2f2f2' },
         }}
       >
-        {categories.map((category, i) => (
-          <MenuItem key={i} value={category.id} sx={{ fontSize: '0.85em', fontWeight: 600 }}>
-            {category.name}
-          </MenuItem>
-        ))}
+        {categories.map((category, i) => {
```
Comment (avirankatz) [inline frontend/src/components/CategorySelect.tsx:124]: We should use the old mapping function (the one which is present in dev, and add another menu item below it for new category

File: frontend/src/contexts/group.tsx (hunk 1 above)
Comment (avirankatz) [reply frontend/src/contexts/group.tsx:72]: Also, the string "New Category" should be dynamic, implemented in `bank.ts` and `en.ts` (see https://github.com/consenz-io/webapp/blob/main/frontend/README.md#31-strings)

**Review by avirankatz — CHANGES_REQUESTED** [review]

File: frontend/src/contexts/group.tsx (hunk 1 above)
Comment (yishain11) [reply frontend/src/contexts/group.tsx:72]: both fixed

File: frontend/src/contexts/group.tsx (hunk 2 above)
Comment (yishain11) [reply frontend/src/contexts/group.tsx]: done

File: frontend/src/pages/NewAgreement.tsx (hunk 3 above)
Comment (yishain11) [reply frontend/src/pages/NewAgreement.tsx:65]: removed

File: frontend/src/components/CategorySelect.tsx (hunk 4 above)
Comment (yishain11) [reply frontend/src/components/CategorySelect.tsx:124]: done

File: frontend/src/components/CategorySelect.tsx [hunk 5]
```diff
@@ -112,11 +120,27 @@ const CategorySelect: FC<ICategorySelectProps> = ({
...
+            fontSize: '0.85em',
+            fontWeight: 600,
+          }}
+          key={NaN}
+        >
+          <ListItemIcon sx={{ margin: 0 }}>
+            <AddIcon fontSize="small" />
+          </ListItemIcon>
+          {t(StringBank.ADD_NEW_CATEGORY)}
+        </MenuItem>
```
Comment (avirankatz) [inline frontend/src/components/CategorySelect.tsx:143]:
```suggestion
        {categories.map((category, i) => (
          <MenuItem key={i} value={category.id} sx={{ fontSize: '0.85em', fontWeight: 600 }}>
            {category.name}
          </MenuItem>
        ))}
        <MenuItem>
          <AddIcon fontSize="small" htmlColor="#adb2b8" />
          <Typography variant="body2" color="#adb2b8" paddingX={1}>
            {t(StringBank.ADD_NEW_CATEGORY)}
          </Typography>
        </MenuItem>
```

**Review by avirankatz — CHANGES_REQUESTED** [review]

File: frontend/src/components/CategorySelect.tsx (hunk 5 above)
Comment (avirankatz) [reply frontend/src/components/CategorySelect.tsx:143]: Short explanation for the suggestion:
1. we don't need to use a `key` prop outside a loop
2. We already have a typography variant that implements font size of 0.875rem (or 14px), and normal weight, which is body2. So I used it and added some coloring and padding according to the zeplin.

**Review by avirankatz — APPROVED** [review]

**Pull request merged**